src/model/model.nit

```diff
@@ -1095,6 +1113,46 @@ abstract class MType
 	# ENSURE: `not self.need_anchor implies result == true`
 	fun can_resolve_for(mtype: MType, anchor: nullable MClassType, mmodule: MModule): Bool is abstract
 
+	# Given the specified anchor, does this type allows `null`?
+	fun can_be_null(mmodule: MModule, anchor: nullable MClassType): Bool
+	do
+		return false
+	end
+
+	# Intersect type with the specified type.
+	#
+	# The resulting type represents the subtypes that are common to both `self`
+	# and `other`. The contextual parameters (`mmodule` and `anchor`) are
+	# used to simplify the resulting type.
```

Comment:
Right OK, they are used to resolve in order to simplify. The doc mention optimizing types somewhere else in this PR, is "simplify" and "optimize" the same concept?